### DISCUSSION TITLE
Tasmota_types.h  --> TSettings

### DISCUSSION THREAD

**Discussion by trlafleur** [arendst/Tasmota#15791] [Q&A] [answered]
I've noticed that TSettings is running out of its 4096 allocation block.... what are the plans for the future???

It would be nice when more space is allocated to have a dedicated block for drive development, where one would not need to squeeze into the current block.

thanks...
trying to move my code to 11.1.0.4

**Comment by s-hadinger** [comment, marked as answer]
We tend to disregard such a generic approach until we have a definite use case. Finding a new 4kb block is not straightforward. 

What we suggest you do is use the file system for additional information that can't fit settings. This is what I do with Zigbee. 

There is also the persist module in Berry for that purpose

**Reply by arendst** [reply]
Indeed. Filesystem is the way forward.

I used the latest technics in driver xdrv_86_esp32_sonoff_spm.ino which uses the file system to store any kind of variable structure you like. It is serviced equal like the current settings area by transparently storing and updating your self-defined structure while using the same consistent save on restart on save on change.

Before I introduced a demo driver xdrv_98_file_settings_demo.ino showing the basics. As the SPM is in production I suggest to base your driver on this one.

**Reply by sfromis** [reply]
For configuration data not fitting in the 4K settings block, another plan could be to have them initialized each boot, allowing the user to select how, including from `autoexec.bat`.
